app/controllers/tags_controller.rb

```diff
@@ -30,13 +41,32 @@ def show
       ").
       select('vulnerability_tags.note as tag_note').
       order('vulnerabilities.upvotes desc')
-    render_json_for_api(@vulns)
+    if @vulns == nil
+      flash[:error] = "Error: Tag not found"
+      redirect_to controller: 'tags', :action => 'index'
+    else
+      render_json_for_api(@vulns)
+    end
   end
 
   private
     # Use callbacks to share common setup or constraints between actions.
     def set_tag
-      @tag = Tag.find(params[:id])
+      if(params[:id].match(/^(\d)+$/).present?)
```

Comment:
Looks good, but could be DRYer. There's a shorter way to do this:

```suggestion
      key = params[:id].match(/^(\d)+$/).present? ? "id" : "shortname"
      @tag = Tag.find(key: params[:id])
      # and then you only have one begin/rescue loop
               
```

Reply:
and by "loop" I mean begin/rescue "clause"